query/storage.go

```diff
@@ -96,6 +96,7 @@ type ReadWindowAggregateSpec struct {
 	Aggregates  []plan.ProcedureKind
 	CreateEmpty bool
 	TimeColumn  string
+	Window      execute.Window
```

Comment:
Why do we need this? Does it duplicate `WindowEvery` and `Offset` in this struct?  I don't see the value of this gets assigned anywhere in the code?

Reply:
Oh, yeah that's a remnant from the duration support update. We will need it in the future for when we add monthly durations, but with some revision I can remove it for now and we can add it back when we make that change.

Reply:
And for the record, it was used in the added code. it was used to create durations like here: https://github.com/influxdata/influxdb/blob/f34cee7199384e52eaf8dd98664ab072883395a4/storage/flux/reader.go#L278-L300

But I'll be removing that since the duration support update was not an intentional part of this PR.

Reply:
I'm fine if we keep this since it's just going to be added anyway when we enable the rest of the pushdowns.

Reply:
We don't actually have a current issue for adding duration support right now I don't think.

Reply:
Or are you saying that as we port over all the existing code, we'll get this revision for free as long as we don't remove these bits?

Reply:
Yes that's correct.

Reply:
I'm fine with leaving it then. @yzhang1991 ?

Reply:
@jlapacik in that case, I think that leaving duration support in place and enabling those tests should be added to the DOD for finalizing the other pushdowns.